base_layer/core/src/base_node/sync/header_sync/synchronizer.rs

```diff
@@ -462,21 +487,21 @@ impl<'a, B: BlockchainBackend + 'static> HeaderSynchronizer<'a, B> {
         );
 
         // Basic sanity check that the peer sent tip height greater than the split.
-        let split_height = local_tip_header.height().saturating_sub(steps_back);
-        if remote_tip_height < split_height {
+        let split_height = best_header.height().saturating_sub(peer_response.reorg_steps_back);
+        if peer_response.remote_tip_height < split_height {
             return Err(BlockHeaderSyncError::InvalidProtocolResponse(format!(
```

Comment:
Does this result in a ban? I think it should

Reply:
Yes it does